### ISSUE TITLE
Incorrect error message when running "zwe install" command without node variable defined

### ISSUE THREAD

**Issue by js665999** [zowe/zowe-install-packaging#2839]
**Describe the bug**
Incorrect error message when running "zwe install" command without node.home defined in zowe.yaml. Message asks to define NODE_HOME environment variable instead. Zowe v2.0.0

**To Reproduce**
1. define NODE_HOME environment variable
2. do NOT fill node.home variable in zowe.yaml
3. run "zwe install"

**Error message shown**
Error ZWEL0121E: Cannot find node. Please define NODE_HOME environment variable.

**Expected behavior**
Message showing correct location of variable to update.
OR
Using node from NODE_HOME

**Comment by jackjia-ibm** [comment]
@js665999 I couldn't reproduce this issue on my local system.

Are you sure your NODE_HOME points correct location? I saw one common issue could be it points to node bin directory, but actually shouldn't have bin in the path. Please let me know.

**Comment by js665999** [comment]
@jackjia-ibm I confirmed the path is correct. 

![Capture](https://user-images.githubusercontent.com/45845701/167097061-647ea1ba-fdeb-4cfe-b465-98c936579cec.PNG)

If I use identical node path in zowe.yaml it works 
I tried different ways to set the variable before the command to ensure it sees it. Same result.

**I just tried to reproduce on different system and was unable to do so.** So it seems the problem is with something on this specific system.

**Comment by jackjia-ibm** [comment]
@js665999 it's still strange for that particular system. Not sure if you try the trace mode by passing `--trace` or `-vv` to `zwe install` and see if provides more information?

**Comment by js665999** [comment]
@jackjia-ibm 
I did some digging in the scripts and I believe I found the issue. There is a function **require_node** which does not check for NODE_HOME variable before rewriting it with contents of yaml file:

-  **shell_read_yaml_node_home** sets NODE_HOME to empty if node.home not set in zowe.yaml
- **detect_node_home** does not work on affected system(NODE_HOME  already overwritten, _which_ command not available and node was not in PATH), however works on other system. This is the reason for not being able to reproduce.
- **print_error_and_exit** exits without ever checking original value of $NODE_HOME

```
<runtime_dir>/bin/libs/node.sh:82
require_node() {
  # prepare the NODE_HOME in zowe.yaml
  if [ -n "${ZWE_CLI_PARAMETER_CONFIG}" ]; then
    export NODE_HOME=$(shell_read_yaml_node_home "${ZWE_CLI_PARAMETER_CONFIG}") 
  fi
  if [ -z "${NODE_HOME}" ]; then
    export NODE_HOME=$(detect_node_home) 
  fi

  if [ -z "${NODE_HOME}" ]; then
    print_error_and_exit "Error ZWEL0121E: Cannot find node. Please define NODE_HOME environment variable." "" 121 
  fi

  ensure_node_is_on_path
}

```

**Comment by jackjia-ibm** [comment]
Thanks a lot @js665999 for digging into code and found the root cause. I will put a fix on this and release in v2.1.

**Comment by jackjia-ibm** [comment]
Please re-open if the issue is not fixed in v2.1.0 release. Thanks.
